br/pkg/storage/parse.go

```diff
@@ -207,6 +208,31 @@ func FormatBackendURL(backend *backuppb.StorageBackend) (u url.URL) {
 	return
 }
 
+// RedactURL redacts the secret tokens in the URL. only S3 url need redaction for now.
+func RedactURL(str string) (string, error) {
+	u, err := ParseRawURL(str)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	scheme := u.Scheme
+	failpoint.Inject("forceRedactURL", func() {
+		scheme = "s3"
+	})
+	if strings.ToLower(scheme) == "s3" {
```

Comment:
How about defining some constants for "s3", "access-key", "secret-access-key"?

Reply:
there're many such string literal in this package, will not do it in this pr

Reply:
Sure~